From cb021ca8e2d6f4d457de5011ea9a939db84ab5a2 Mon Sep 17 00:00:00 2001
From: Francesco Pham <francesco.pham1@h-partners.com>
Date: Thu, 28 May 2026 12:00:41 +0200
Subject: [PATCH 2/2] asmjit: add OH GN BUILD and bundle.json

Register asmjit as an OH 'thirdparty' subsystem component with two targets: libasmjit (shared, runtime) and libasmjit_frontend_static. Public config exports the diagnostic pragmas and -I src so consumers compile cleanly. Listed last on purpose in arkencoder_config: GN appends config cflags in order, and the -Wno-bitwise-instead-of-logical / -Wno-unused-but-set-variable / -Wno-deprecated-copy here must come after ark_config's -Wall to win.

Signed-off-by: Francesco Pham <francesco.pham1@h-partners.com>
---
 BUILD.gn    | 171 ++++++++++++++++++++++++++++++++++++++++++++++++++++
 bundle.json |  43 +++++++++++++
 2 files changed, 214 insertions(+)
 create mode 100644 BUILD.gn
 create mode 100644 bundle.json

diff --git a/BUILD.gn b/BUILD.gn
new file mode 100644
index 0000000..744d48d
--- /dev/null
+++ b/BUILD.gn
@@ -0,0 +1,171 @@
+# Copyright (c) 2021-2024 Huawei Device Co., Ltd.
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import("//arkcompiler/runtime_core/static_core/ark_config.gni")
+import("//build/ohos.gni")
+
+config("asmjit_public_config") {
+    include_dirs = [
+        "$ark_third_party_root/asmjit/src",
+    ]
+    cflags_cc = [
+        "-Wno-bitwise-instead-of-logical",
+        "-Wno-unused-but-set-variable",
+        "-Wno-deprecated-copy",
+    ]
+    if (ark_static_standalone_build) {
+        ldflags = [ "-lrt" ]
+    }
+}
+
+asmjit_sources = [
+    "src/asmjit/asmjit.h",
+
+    "src/asmjit/core.h",
+    "src/asmjit/core/api-build_p.h",
+    "src/asmjit/core/api-config.h",
+    "src/asmjit/core/arch.cpp",
+    "src/asmjit/core/arch.h",
+    "src/asmjit/core/assembler.cpp",
+    "src/asmjit/core/assembler.h",
+    "src/asmjit/core/builder.cpp",
+    "src/asmjit/core/builder.h",
+    "src/asmjit/core/callconv.cpp",
+    "src/asmjit/core/callconv.h",
+    "src/asmjit/core/codebuffer.h",
+    "src/asmjit/core/codebufferwriter_p.h",
+    "src/asmjit/core/codeholder.cpp",
+    "src/asmjit/core/codeholder.h",
+    "src/asmjit/core/compiler.cpp",
+    "src/asmjit/core/compiler.h",
+    "src/asmjit/core/constpool.cpp",
+    "src/asmjit/core/constpool.h",
+    "src/asmjit/core/cpuinfo.cpp",
+    "src/asmjit/core/cpuinfo.h",
+    "src/asmjit/core/datatypes.h",
+    "src/asmjit/core/emitter.cpp",
+    "src/asmjit/core/emitter.h",
+    "src/asmjit/core/emitterutils.cpp",
+    "src/asmjit/core/emitterutils_p.h",
+    "src/asmjit/core/environment.cpp",
+    "src/asmjit/core/environment.h",
+    "src/asmjit/core/errorhandler.cpp",
+    "src/asmjit/core/errorhandler.h",
+    "src/asmjit/core/features.h",
+    "src/asmjit/core/formatter.cpp",
+    "src/asmjit/core/formatter.h",
+    "src/asmjit/core/func.cpp",
+    "src/asmjit/core/func.h",
+    "src/asmjit/core/globals.cpp",
+    "src/asmjit/core/globals.h",
+    "src/asmjit/core/inst.cpp",
+    "src/asmjit/core/inst.h",
+    "src/asmjit/core/jitallocator.cpp",
+    "src/asmjit/core/jitallocator.h",
+    "src/asmjit/core/jitruntime.cpp",
+    "src/asmjit/core/jitruntime.h",
+    "src/asmjit/core/logger.cpp",
+    "src/asmjit/core/logger.h",
+    "src/asmjit/core/misc_p.h",
+    "src/asmjit/core/operand.cpp",
+    "src/asmjit/core/operand.h",
+    "src/asmjit/core/osutils.cpp",
+    "src/asmjit/core/osutils.h",
+    "src/asmjit/core/raassignment_p.h",
+    "src/asmjit/core/rabuilders_p.h",
+    "src/asmjit/core/radefs_p.h",
+    "src/asmjit/core/ralocal.cpp",
+    "src/asmjit/core/ralocal_p.h",
+    "src/asmjit/core/rapass.cpp",
+    "src/asmjit/core/rapass_p.h",
+    "src/asmjit/core/rastack.cpp",
+    "src/asmjit/core/rastack_p.h",
+    "src/asmjit/core/string.cpp",
+    "src/asmjit/core/string.h",
+    "src/asmjit/core/support.cpp",
+    "src/asmjit/core/support.h",
+    "src/asmjit/core/target.cpp",
+    "src/asmjit/core/target.h",
+    "src/asmjit/core/type.cpp",
+    "src/asmjit/core/type.h",
+    "src/asmjit/core/virtmem.cpp",
+    "src/asmjit/core/virtmem.h",
+    "src/asmjit/core/zone.cpp",
+    "src/asmjit/core/zone.h",
+    "src/asmjit/core/zonehash.cpp",
+    "src/asmjit/core/zonehash.h",
+    "src/asmjit/core/zonelist.cpp",
+    "src/asmjit/core/zonelist.h",
+    "src/asmjit/core/zonestack.cpp",
+    "src/asmjit/core/zonestack.h",
+    "src/asmjit/core/zonestring.h",
+    "src/asmjit/core/zonetree.cpp",
+    "src/asmjit/core/zonetree.h",
+    "src/asmjit/core/zonevector.cpp",
+    "src/asmjit/core/zonevector.h",
+
+    "src/asmjit/x86.h",
+    "src/asmjit/x86/x86archdata.cpp",
+    "src/asmjit/x86/x86archdata_p.h",
+    "src/asmjit/x86/x86assembler.cpp",
+    "src/asmjit/x86/x86assembler.h",
+    "src/asmjit/x86/x86builder.cpp",
+    "src/asmjit/x86/x86builder.h",
+    "src/asmjit/x86/x86callconv.cpp",
+    "src/asmjit/x86/x86callconv_p.h",
+    "src/asmjit/x86/x86compiler.cpp",
+    "src/asmjit/x86/x86compiler.h",
+    "src/asmjit/x86/x86emitter.h",
+    "src/asmjit/x86/x86features.cpp",
+    "src/asmjit/x86/x86features.h",
+    "src/asmjit/x86/x86formatter.cpp",
+    "src/asmjit/x86/x86formatter_p.h",
+    "src/asmjit/x86/x86globals.h",
+    "src/asmjit/x86/x86internal.cpp",
+    "src/asmjit/x86/x86internal_p.h",
+    "src/asmjit/x86/x86instdb.cpp",
+    "src/asmjit/x86/x86instdb.h",
+    "src/asmjit/x86/x86instdb_p.h",
+    "src/asmjit/x86/x86instapi.cpp",
+    "src/asmjit/x86/x86instapi_p.h",
+    "src/asmjit/x86/x86operand.cpp",
+    "src/asmjit/x86/x86operand.h",
+    "src/asmjit/x86/x86rapass.cpp",
+    "src/asmjit/x86/x86rapass_p.h",
+]
+
+asmjit_configs = [
+    "$ark_root:ark_config",
+    "$ark_root/libarkbase:arkbase_public_config",
+
+    # Listed last on purpose: GN appends config cflags in order, and our
+    # -Wno-bitwise-instead-of-logical/-Wno-unused-but-set-variable/
+    # -Wno-deprecated-copy here must come after ark_config's -Wall to win.
+    ":asmjit_public_config",
+]
+
+ohos_static_library("libasmjit") {
+    sources = asmjit_sources
+    configs = asmjit_configs
+    deps = [ "$ark_root/libarkbase:libarktsbase" ]
+    part_name = "asmjit"
+    subsystem_name = "thirdparty"
+}
+
+ohos_static_library("libasmjit_frontend_static") {
+    sources = asmjit_sources
+    configs = asmjit_configs
+    deps = [ "$ark_root/libarkbase:libarktsbase_frontend_static" ]
+    part_name = "asmjit"
+    subsystem_name = "thirdparty"
+}
diff --git a/bundle.json b/bundle.json
new file mode 100644
index 0000000..db9f71c
--- /dev/null
+++ b/bundle.json
@@ -0,0 +1,43 @@
+{
+    "name": "@ohos/asmjit",
+    "description": "AsmJit is a lightweight library for machine code generation written in C++.",
+    "version": "1.0",
+    "license": "Zlib",
+    "publishAs": "code-segment",
+    "segment": {
+        "destPath": "third_party/asmjit"
+    },
+    "dirs": {},
+    "scripts": {},
+    "licensePath": "LICENSE.md",
+    "component": {
+        "name": "asmjit",
+        "subsystem": "thirdparty",
+        "syscap": [],
+        "features": [],
+        "adapted_system_type": [
+            "standard"
+        ],
+        "rom": "",
+        "ram": "",
+        "deps": {
+            "components": [],
+            "third_party": []
+        },
+        "build": {
+            "sub_component": [
+                "//third_party/asmjit:libasmjit",
+                "//third_party/asmjit:libasmjit_frontend_static"
+            ],
+            "inner_kits": [
+                {
+                    "name": "//third_party/asmjit:libasmjit"
+                },
+                {
+                    "name": "//third_party/asmjit:libasmjit_frontend_static"
+                }
+            ],
+            "test": []
+        }
+    }
+}
-- 
2.50.1

